### DISCUSSION TITLE
Update History in single file report mode

### DISCUSSION THREAD

**Discussion by tuscias** [allure-framework/allure2#2127] [Questions & Support]
Hi, 
I was wondering if single page report mode supports history updates?
Thanks

**Comment by baev** [comment]
Not yet, but we plan to fix it.

**Comment by storenth** [comment]
@baev seems like including the history in a single report file is more of a logical issue rather than a technical issue?

**Reply by baev** [reply]
Yes, you're right. I do believe users expect a single file to be the output if the `--single-file` option is used. So we either should introduce a separate command to generate the history or maybe add another option to produce the history (`--history path/to/history/folder`).

**Reply by sergs-pci** [reply]
@storenth @tuscias 
As for now, I'm using it as a workaround to run report generation twice
Something like that:
```
cp ./last-history/* ./allure-results/history
allure generate -c ./allure-results -o ./allure-report-history
allure generate -c --single-file ./allure-results -o ./allure-report
cp ./allure-report-history/history/* ./last-history

```
will create for you the single file ./allure-report/index.html with embedded trends&history

**Reply by Hydrair** [reply]
> @storenth @tuscias As for now, I'm using it as a workaround to run report generation twice Something like that:
> 
> ```
> cp ./last-history/* ./allure-results/history
> allure generate -c ./allure-results -o ./allure-report-history
> allure generate -c --single-file ./allure-results -o ./allure-report
> cp ./allure-report-history/history/* ./last-history
> ```
> 
> will create for you the single file ./allure-report/index.html with embedded trends&history

I tried that too. With an executor.json it creates a history with links to the older test results. But these links lead to error 404 as the ids are apparently different. 
For example a link in `reports/936426/#suites/c99683b02220aed66410547786677cb1/8aceac9b2c2b4b09/history` leads to `reports/936385/#testresult/eb541f1283542eb5`, which results in 404. The correct ID in the older run would be `reports/936385/#testresult/145b4f21a9c8ec85`.

Is there a way to preserve IDs when creating reports multiple times?

**Reply by DogeTheBeast** [reply]
I am stuck with the same issue. The ideal solution would be the `--history` flag but I am not sure if this is even a planned feature or under development. 
Any updates on this issue ?
